pkg/serverless/daemon/routes_test.go

```diff
@@ -45,7 +45,9 @@ func TestStartInvocation(t *testing.T) {
 	assert.Nil(err)
 	res, err := client.Do(request)
 	assert.Nil(err)
-	assert.Equal(res.StatusCode, 200)
+	if res != nil {
```

Comment:
It seems like the response can be nil even when there isn't an error, is this expected?

Reply:
The thing is, after an assertion fails the test actually keeps running. So, we can't assume that res != nil, we have to check it. Otherwise, we might try to look up the StatusCode on a nil request.

Reply:
Could we use [require](https://github.com/DataDog/datadog-agent/blob/main/test/integration/config_providers/zookeeper/zookeeper_provider_test.go#L143) for that instead? I believe it's also used in a lot of other places in the datadog-agent repo.

Reply:
I guess my question was, _why_ are both `res` and `err` nil? It doesn't seem like this should be the case?

Reply:
`require` is cool, that would work well here. 👍 

In this case, `err` is not nil but `res` is nil. We then check `res.StatusCode` and the test panics.

Reply:
Ah gotcha, sounds good!